application/src/main/java/org/opentripplanner/street/search/request/TimeSlopeSafetyTriangle.java

```diff
@@ -0,0 +1,143 @@
+package org.opentripplanner.street.search.request;
+
+import static org.opentripplanner.utils.lang.DoubleUtils.doubleEquals;
+import static org.opentripplanner.utils.lang.DoubleUtils.roundTo2Decimals;
+
+/**
+ * Sets the (bicycle or scooter) triangle routing parameters -- the relative importance of safety,
+ * flatness, and speed. These three fields should have values between 0 and 1, and should add up to
+ * 1.
+ * <p>
+ * The constructor accepts any three numbers and will normalize them to add up to 1. {@code time}
+ * and {@code slope} are rounded to the closest two decimal number, then
+ * {@code safety := 1.0 - (time + slope)}. This is done to make the rounding predictable and to
+ * allways add up to one. This allows this class to be used in an index of a cache. For example:
+ * <pre>
+ *   ( 1.0, 1.0, 1.0 ) => ( time: 0.33, slope: 0.33, safety: 0.34 )
+ * </pre>
+ * <p>
+ * This class is currently only used with bicycle routing, but is not limited to that.
+ * <p>
+ * THIS CLASS IS IMMUTABLE AND THREAD-SAFE.
+ */
+public record TimeSlopeSafetyTriangle(double time, double slope, double safety) {
```

Comment:
- No records in the domain layer.
- This is a smal simple value object; hence should only return a compact version of it self in the `toString` method. Either: `[0.7, 0.25, 0.05 ]` or `[time: 0.7, slope: 0.33, safety: 0.34]`. Simple value objects never exist by them self so the context is enough to tell the type.